src/main/java/zmq/Ctx.java

```diff
@@ -127,6 +134,13 @@ private void destroy() throws IOException
             it.close();
         }
 
+        for (Selector selector : selectors) {
+            if (selector != null) {
```

Comment:
It does no harm to leave the checking here. In #230, the null checking is to avoid getting a NPE, as the selector is a member of the class that can be nullified (basically calling 2 or more times close() should be done without side-effects).
The call to Selector.open() is the only way to add one into the list, so if this method is trusted to always deliver an instance, there is no need to do so. As you stated @colini, an assert(selector != null) should suffice.
Here there is the liberty to choose which defensive programming you want to apply, in one case the path is to close and release resources, in the other it is about allocating the resources, both stages where performance may not be the most important point.